integrations/watsonx/src/haystack_integrations/components/embedders/watsonx/document_embedder.py

```diff
@@ -0,0 +1,196 @@
+        self.truncate_input_tokens = truncate_input_tokens
+        self.prefix = prefix
+        self.suffix = suffix
+        self.batch_size = batch_size
+        self.concurrency_limit = concurrency_limit
+        self.timeout = timeout
+        self.max_retries = max_retries
+
+        # Initialize the embeddings client
+        credentials = Credentials(api_key=api_key.resolve_value(), url=url)
+
+        params = {}
+        if truncate_input_tokens is not None:
+            params['truncate_input_tokens'] = truncate_input_tokens
+
+        self.embedder = Embeddings(
+            model_id=model,
+            credentials=credentials,
+            project_id=project_id,
+            space_id=space_id,
+            params=params if params else None,
+            batch_size=batch_size,
+            concurrency_limit=concurrency_limit,
+            max_retries=max_retries or 10,
+        )
+
+    def _get_telemetry_data(self) -> dict[str, Any]:
+        """
+        Data that is sent to Posthog for usage analytics.
+        """
+        return {'model': self.model}
+
+    def to_dict(self) -> dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+        """
+        return default_to_dict(
+            self,
+            model=self.model,
+            api_key=self.api_key.to_dict(),
+            url=self.url,
+            project_id=self.project_id,
+            space_id=self.space_id,
+            truncate_input_tokens=self.truncate_input_tokens,
+            prefix=self.prefix,
+            suffix=self.suffix,
+            batch_size=self.batch_size,
+            concurrency_limit=self.concurrency_limit,
+            timeout=self.timeout,
+            max_retries=self.max_retries,
+        )
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> WatsonXDocumentEmbedder:
+        """
+        Deserializes the component from a dictionary.
+        """
+        deserialize_secrets_inplace(data['init_parameters'], keys=['api_key'])
+        return default_from_dict(cls, data)
+
+    def _prepare_text(self, text: str) -> str:
+        """
+        Prepares text for embedding by adding prefix and suffix.
+        """
+        return self.prefix + text + self.suffix
+
+    @component.output_types(documents=List[Document], meta=Dict[str, Any])
+    def run(self, documents: list[Document]):
+        """
+        Embeds a list of documents.
+
+        :param documents:
+            A list of documents to embed.
+        :returns:
+            A dictionary with:
+            - 'documents': List of Documents with embeddings added
+            - 'meta': Information about the model usage
+        """
+        if not isinstance(documents, list) or (documents and not isinstance(documents[0], Document)):
+            msg = (
+                'WatsonXDocumentEmbedder expects a list of Documents as input.'
+                'In case you want to embed a string, please use the WatsonXTextEmbedder.'
```

Comment:
Add space otherwise two sentences are concatenated with no space
```suggestion
                ' In case you want to embed a string, please use the WatsonXTextEmbedder.'
```